inspectors/utils/utils.py

```diff
@@ -420,6 +423,37 @@ def check_tool_present(*args):
   _tool_present_cache[args] = result
   return result
 
+# read PDF's directory to determine if we need to decrypt it
+def check_pdf_decryption(pdf_path):
+  try:
+    doc = pdfrw.PdfReader(pdf_path)
+    return "/Encrypt" in doc
```

Comment:
This is PDF internals stuff, the line is checking for whether there's an "encryption dictionary" in the document's "trailer dictionary." Presence or absence of this determines whether we run `qpdf --decrypt` on it later.